### PR TITLE
Add support for inlined enum variants

### PR DESCRIPTION
**Pull request by JMLX42**
Implementation for #727

The following is now supported:

```rust
    #[derive(ToSchema)]
    enum Number {
        One,
        Two,
        Three,
        Four,
        Five,
        Six,
        Seven,
        Height,
        Nine,
    }

    #[derive(ToSchema)]
    enum Color {
        Spade,
        Heart,
        Club,
        Diamond,
    }

    #[derive(ToSchema)]
    enum Card {
        Number(#[schema(inline)] Number),
        Color(#[schema(inline)] Color),
    }
```

and will produce the following schema:

```json
{
            "oneOf": [
                {
                    "properties": {
                        "Number": {
                            "enum": [
                                "One",
                                "Two",
                                "Three",
                                "Four",
                                "Five",
                                "Six",
                                "Seven",
                                "Height",
                                "Nine",
                            ],
                            "type": "string",
                        },
                    },
                    "required": [
                        "Number",
                    ],
                    "type": "object",
                },
                {
                    "properties": {
                        "Color": {
                            "enum": [
                                "Spade",
                                "Heart",
                                "Club",
                                "Diamond",
                            ],
                            "type": "string",
                        },
                    },
                    "required": [
                        "Color",
                    ],
                    "type": "object",
                },
            ],
        }
```

### CODE REVIEW

File: utoipa-gen/src/component/schema.rs [hunk 1]
```diff
@@ -619,6 +619,14 @@ impl ToTokensDiagnostics for UnnamedStructSchema<'_> {
 
             if fields_len == 1 {
                 if let Some(ref mut features) = unnamed_struct_features {
+                    let inline =
+                        features::parse_schema_features_with(&first_field.attrs, |input| {
```
Comment (JMLX42) [inline utoipa-gen/src/component/schema.rs:623]: @juhaku that looks kinda naive. Isn't there a way to properly specify which features are supported by this specific case?

Comment (JMLX42) [reply utoipa-gen/src/component/schema.rs:623]: What I mean is I'd rather have something like this: https://github.com/juhaku/utoipa/blob/d020f92a0446adc7e99999aa9e1be63e95728fbb/utoipa-gen/src/component/schema/features.rs#L150

But those features are for the variant themselves, so `Inline` would be:

```rust
    #[derive(ToSchema)]
    enum Card {
        #[schema(inline)]
        Number(Number),
        #[schema(inline)]
        Color(Color),
    }
```

Which IMO does not fit with the rest of the API and is counter-intuitive.

But I haven't found the features struct for the unnamed field.

Comment (juhaku) [reply utoipa-gen/src/component/schema.rs:623]: > @juhaku that looks kinda naive. Isn't there a way to properly specify which features are supported by this specific case?

No features have been originally defined for unnamed field enum variants itself thus there is no such struct. Though if there is just a single feature to be parsed the `parse_with` is just fine but if there would be multiple features or the set of features is known to be needed in multiple locations then a feature struct is preferred. 

> Which IMO does not fit with the rest of the API and is counter-intuitive.


Yeah, I agree the API would be counter intuitive and the features defined directly for the unnamed enum variant field is cleaner approach. 

> But I haven't found the features struct for the unnamed field.

Such struct does not exists by design but could be added if needed.

**Review by juhaku — CHANGES_REQUESTED** [review]
This is good for me :tada: 

Though docs should be updated with the added feature at `ToSchema` macro by changing the doc comment here https://github.com/juhaku/utoipa/blob/d020f92a0446adc7e99999aa9e1be63e95728fbb/utoipa-gen/src/lib.rs#L58.

**Comment by JMLX42** [comment]
> This is good for me 🎉
> 
> Though docs should be updated with the added feature at `ToSchema` macro by changing the doc comment here
> 
> https://github.com/juhaku/utoipa/blob/d020f92a0446adc7e99999aa9e1be63e95728fbb/utoipa-gen/src/lib.rs#L58
> .

@juhaku I guess [this section](https://github.com/juhaku/utoipa/blob/d020f92a0446adc7e99999aa9e1be63e95728fbb/utoipa-gen/src/lib.rs#L136) would be the right place to document the feature?

**Comment by juhaku** [comment]
> @juhaku I guess [this section](https://github.com/juhaku/utoipa/blob/d020f92a0446adc7e99999aa9e1be63e95728fbb/utoipa-gen/src/lib.rs#L136) would be the right place to document the feature?

Yes that chapter would be the right place for the doc. It should describe that it is used directly for the variant field and not for the variant itself. And perhaps an example of this would be good to add as well under the examples section a little bit further down.

**Pull request merged**